eppo_client/client.py

```diff
@@ -227,3 +236,20 @@ def check_type_match(
     expected_type: Optional[VariationType], actual_type: VariationType
 ):
     return expected_type is None or actual_type == expected_type
+
+
+def check_value_type_match(
+    expected_type: Optional[VariationType], value: ValueType
+) -> bool:
+    if expected_type is None:
+        return True
+    if expected_type in [VariationType.JSON, VariationType.STRING]:
+        return isinstance(value, str)
+    if expected_type == VariationType.INTEGER:
+        return isinstance(value, int)
+    if expected_type == VariationType.NUMERIC:
+        # we can convert int to float
+        return isinstance(value, float) or isinstance(value, int)
```

Comment:
`or isinstance(value, int)` seems unnecessary? With this PR we _always_ convert to float? Other than in one of the tests where I see that you explicitly rely on this behavior. In any case this is fine to keep - it's more that I want to check my understanding.

Reply:
Without the `or` this check would fail for an integer value such as `1`, which we can convert to a float

Reply:
I get it now. The cast to `float` is done _after_ validating the value type, so this is indeed necessary.